hypothesis-python/src/hypothesis/control.py

```diff
@@ -149,10 +149,10 @@ def prep_args_kwargs_from_strategies(self, kwarg_strategies):
         arg_labels = {}
         kwargs = {}
         for k, s in kwarg_strategies.items():
-            start_idx = self.data.index_ir
+            start_idx = len(self.data.ir_nodes)
             with deprecate_random_in_strategy("from {}={!r}", k, s) as check:
                 obj = check(self.data.draw(s, observe_as=f"generate:{k}"))
-            end_idx = self.data.index_ir
+            end_idx = len(self.data.ir_nodes)
             kwargs[k] = obj
```

Comment:
this is because `index_ir` has diverged a bit from the buffer concept of "index into the choice sequence". Rather than try to sync a separate variable with that concept like we did with `.index` for buffers, I think the very few usage sites can just use the more verbose `len(nodes)`.